src/app/modules/map-config/services/map-layer-form-builder/map-layer-form-builder.service.ts

```diff
@@ -187,7 +190,32 @@ export class MapLayerAllTypesFormGroup extends ConfigFormGroup {
           false,
           valuesToOptions(geometryTypes),
           {
-            resetDependantsOnChange: true
+            resetDependantsOnChange: true,
+            dependsOn: () => [this.geometryStep],
+            onDependencyChange: (control) => {
+              // Feature Mode and Feature Metric
+              if (!!this.geometryStep.get('geometry')) {
+                if (!!this.geometryStep.get('geometry').value) {
+                  (this.geometryStep.get('geometry') as SelectFormControl).sourceData.subscribe((fields: CollectionField[]) => {
+                    fields.forEach(f => {
+                      if (f.name === this.geometryStep.get('geometry').value) {
+                        f.type === CollectionReferenceDescriptionProperty.TypeEnum.GEOPOINT ?
+                          control.setValue(GEOMETRY_TYPE.circle) : control.setValue(GEOMETRY_TYPE.line);
+                      }
+                    });
+                  });
+                }
+              }
+              // Cluster Mode
+              if (!!this.geometryStep.get('aggregatedGeometry')) {
+                if (!!this.geometryStep.get('aggregatedGeometry').value) {
+                  (this.geometryStep.get('aggregatedGeometry').value === AGGREGATE_GEOMETRY_TYPE.geohash_center ||
+                    this.geometryStep.get('aggregatedGeometry').value === AGGREGATE_GEOMETRY_TYPE.centroid ?
+                    control.setValue(GEOMETRY_TYPE.circle) : control.setValue(GEOMETRY_TYPE.fill));
+                }
```

Comment:
is it possible to handle the `rawGeometry` step and deal with it as a Feature mode ?

Reply:
Done